lib/srv/db/autousers_test.go

```diff
@@ -30,49 +30,81 @@ import (
 // TestAutoUsersPostgres verifies automatic database user creation for Postgres.
 func TestAutoUsersPostgres(t *testing.T) {
 	ctx := context.Background()
-	testCtx := setupTestContext(ctx, t, withSelfHostedPostgres("postgres", func(db *types.DatabaseV3) {
-		db.Spec.AdminUser = &types.DatabaseAdminUser{
-			Name: "postgres",
-		}
-	}))
-	go testCtx.startHandlingConnections()
-
-	// Create user with role that allows user provisioning.
-	_, role, err := auth.CreateUserAndRole(testCtx.tlsServer.Auth(), "alice", []string{"auto"}, nil)
-	require.NoError(t, err)
-	options := role.GetOptions()
-	options.CreateDatabaseUser = types.NewBoolOption(true)
-	role.SetOptions(options)
-	role.SetDatabaseRoles(types.Allow, []string{"reader", "writer"})
-	role.SetDatabaseNames(types.Allow, []string{"*"})
-	err = testCtx.tlsServer.Auth().UpsertRole(ctx, role)
-	require.NoError(t, err)
-
-	// Try to connect to the database as this user.
-	pgConn, err := testCtx.postgresClient(ctx, "alice", "postgres", "alice", "postgres")
-	require.NoError(t, err)
-
-	// Verify user was activated.
-	select {
-	case e := <-testCtx.postgres["postgres"].db.UserEventsCh():
-		require.Equal(t, "alice", e.Name)
-		require.Equal(t, []string{"reader", "writer"}, e.Roles)
-		require.True(t, e.Active)
-	case <-time.After(5 * time.Second):
-		t.Fatal("user not activated after 5s")
-	}
-
-	// Disconnect.
-	err = pgConn.Close(ctx)
-	require.NoError(t, err)
-
-	// Verify user was deactivated.
-	select {
-	case e := <-testCtx.postgres["postgres"].db.UserEventsCh():
-		require.Equal(t, "alice", e.Name)
-		require.False(t, e.Active)
-	case <-time.After(5 * time.Second):
-		t.Fatal("user not deactivated after 5s")
+	for name, tc := range map[string]struct {
+		mode                types.CreateDatabaseUserMode
+		databaseRoles       []string
+		expectConnectionErr bool
+	}{
+		"activate/deactivate users": {
+			mode:                types.CreateDatabaseUserMode_DB_USER_MODE_KEEP,
+			databaseRoles:       []string{"reader", "writer"},
+			expectConnectionErr: false,
+		},
+		"activate/delete users": {
+			mode:                types.CreateDatabaseUserMode_DB_USER_MODE_PREFER_DROP,
+			databaseRoles:       []string{"reader", "writer"},
+			expectConnectionErr: false,
+		},
+		"disabled": {
+			mode:                types.CreateDatabaseUserMode_DB_USER_MODE_OFF,
+			databaseRoles:       []string{"reader", "writer"},
+			expectConnectionErr: true,
+		},
```

Comment:
I'll add a comment to it. But this should fail because the username is absent on the database and not created by the procedure, so it returns an access denied error.